src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasSerializer.java

```diff
@@ -260,6 +290,69 @@ private Object deserializeResource(final Resource resource)
         }
     }
 
+    private Object deserializeProtoResource(final Resource resource, final String fieldName)
+    {
+        final Field field = readField(fieldName);
+        final Class<?> fieldClass = field.getType();
+        Constructor<?> fieldClassConstructor = null;
+
+        // We need to obtain a dummy instance of the field we want to deserialize. We then use this
+        // dummy instance as a handle to get the correct {@link ProtoAdapter}.
+        try
+        {
+            fieldClassConstructor = fieldClass.getDeclaredConstructor();
+        }
+        catch (final Exception exception)
+        {
+            throw new CoreException("Class {} does not implement a nullary constructor",
+                    fieldClass.getName(), exception);
+        }
+        fieldClassConstructor.setAccessible(true);
+
+        Object handle = null;
+        try
+        {
+            handle = fieldClassConstructor.newInstance();
+        }
+        catch (final Exception exception)
+        {
+            throw new CoreException("Failed to create instance of {}", fieldClass.getName(),
+                    exception);
+        }
+
+        ProtoSerializable protoHandle = null;
+        try
+        {
+            protoHandle = (ProtoSerializable) handle;
+        }
+        catch (final ClassCastException exception)
+        {
+            throw new CoreException("{} is not ProtoSerializable", fieldClass.getName(), exception);
+        }
+
+        final ProtoAdapter adapter = protoHandle.getProtoAdapter();
+        final ProtoSerializable deserializedMember = adapter
+                .deserialize(resource.readBytesAndClose());
```

Comment:
This would return a `byte[]` correct? In case the field is big, would it be possible to stream the resource, one byte buffer at a time using an InputStream instead? That would save memory. Only if it is easy though, as the tradeoff might not be worth it..

Reply:
I will look into this. Could definitely be worth it and shouldn't be too difficult.